feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/repository/addAccount/CloudBackupAddMetaAccountRepository.kt

```diff
@@ -0,0 +1,30 @@
+package io.novafoundation.nova.feature_account_impl.data.repository.addAccount
+
+import io.novafoundation.nova.core_db.dao.MetaAccountDao
+import io.novafoundation.nova.feature_account_api.data.events.MetaAccountChangesEventBus
+import io.novafoundation.nova.feature_account_api.data.proxy.ProxySyncService
+import io.novafoundation.nova.feature_account_api.data.repository.addAccount.AddAccountResult
+import io.novafoundation.nova.feature_account_impl.domain.common.mapLocalAccountFromCloudBackup
+import io.novafoundation.nova.feature_account_impl.domain.common.mapLocalChainAccountsFromCloudBackup
+import io.novafoundation.nova.feature_cloud_backup_api.domain.model.CloudBackupWallet
+
+class CloudBackupAddMetaAccountRepository(
+    proxySyncService: ProxySyncService,
+    metaAccountChangesEventBus: MetaAccountChangesEventBus,
+    private val metaAccountDao: MetaAccountDao
+) : BaseAddAccountRepository<CloudBackupAddMetaAccountRepository.Payload>(
+    proxySyncService,
+    metaAccountChangesEventBus
+) {
+
+    class Payload(val cloudBackupWallet: CloudBackupWallet)
+
+    override suspend fun addAccountInternal(payload: Payload): AddAccountResult {
+        val metaAccount = mapLocalAccountFromCloudBackup(payload.cloudBackupWallet, metaAccountDao::nextAccountPosition)
+        val metaId = metaAccountDao.insertMetaAndChainAccounts(metaAccount) {
+            mapLocalChainAccountsFromCloudBackup(it, payload.cloudBackupWallet)
+        }
```

Comment:
We also need to insert account secrets. We can fix it once https://github.com/novasamatech/nova-wallet-android/pull/1467 is merged by utlizing applyBackupDiff